sdk/monitor/monitor-query/src/metricsClient.ts

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { TokenCredential } from "@azure/core-auth";
+import {
+  bearerTokenAuthenticationPolicy,
+  createPipelineFromOptions,
+  PipelineOptions
+} from "@azure/core-http";
+
+import {
+  GetMetricDefinitionsOptions,
+  GetMetricDefinitionsResponse,
+  GetMetricNamespacesOptions,
+  GetMetricNamespacesResponse,
+  QueryMetricsOptions,
+  QueryMetricsResponse
+} from "./models/publicMetricsModels";
+
+import {
+  KnownApiVersion20170501Preview as MetricsApiVersion,
+  MonitorManagementClient as GeneratedMetricsClient
+} from "./generated/metrics/src";
+import {
+  KnownApiVersion20170501Preview as MetricDefinitionsApiVersion,
+  MetricsDefinitionsClient as GeneratedMetricsDefinitionsClient
+} from "./generated/metricsdefinitions/src";
+import {
+  KnownApiVersion20171201Preview as MetricNamespacesApiVersion,
+  MetricsNamespacesClient as GeneratedMetricsNamespacesClient
+} from "./generated/metricsnamespaces/src";
+import {
+  convertRequestForMetrics,
+  convertRequestOptionsForMetricsDefinitions,
+  convertResponseForMetricNamespaces,
+  convertResponseForMetrics,
+  convertResponseForMetricsDefinitions
+} from "./internal/modelConverters";
+
+export interface MetricsClientOptions extends PipelineOptions {
+  /** server parameter */
+  $host?: string;
```

Comment:
prefer `host` for the public shape, you can map it to `$host` internally

Reply:
Removed and consolidated so it's just `endpoint` for both the metric and log client.